Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -469,8 +469,63 @@ namespace OpenInfraPlatform {
 			void convertIfcOpenCrossProfileDef(const EXPRESSReference<typename IfcEntityTypesT::IfcOpenCrossProfileDef>& profileDef,
 				std::vector<std::vector<carve::geom::vector<2>>>& paths) const
 			{
-				throw oip::UnhandledException(profileDef);
+				if (profileDef->expired())
+					throw oip::ReferenceExpiredException(profileDef);
+
+				std::vector<carve::geom::vector<2>> outer_loop;
+
+				bool horizontal = profileDef->HorizontalWidths;
+
+				std::vector<double> widths;
+				widths.reserve(profileDef->Widths.size());
+				std::transform(
+					profileDef->Widths.begin(),
+					profileDef->Widths.end(),
+					widths.begin(),
+					[](auto& it) {return it * UnitConvert()->getLengthInMeterFactor(); });
+
+				std::vector<double> slopes;
+				slopes.reserve(profileDef->Slopes.size());
+				std::transform(
+					profileDef->Slopes.begin(),
+					profileDef->Slopes.end(),
+					slopes.begin(),
+					[](auto& it) {return it * UnitConvert()->getAngleInRadianFactor(); });
+
+				if (widths.size() != slopes.size()) {
+					throw oip::InconsistentModellingException(profileDef, "Number of Widths is not equal to the number of Slopes");
+				}
+				double TagX = 0.0;
+				double TAgY = 0.0;
+				double x, y;
+
+				for (int i = 0; i < widths.size(); i++) {
+					paths.push_back(carve::geom::VECTOR(TagX, TagY));
+					std::tie(x, y) = CalculateXYFromPolar(horizontal, widths[i], slopes[i]);
+					TagX = TagX + x;
+					TagY = TagY + y;
+				}
+				paths.push_back(carve::geom::VECTOR(TagX, TagY));
+			}
+
+			/*! \brief Calculates X and Y coordinates using slope and width 
+			* \param[in] horizontal A bool variable which defines, if width shall be measured horizontally or along the slope
+			* \param[in] width A double which indicates the horizontal width or distance along the slope for the segment in the profile
+			* \param[in] slope A double which indicates slope measure 
+			* \param[out] x,y X and Y coordinates of the slope end

```

Comment:
This is a `return`, not a `param[out]`. Also, include a reference saying that (0,0) lies at the beginning of the vector.

Reply:
Resolved: 5c1d3b6